### PR TITLE
fix(tableau): miscellaneous tableau fixes for lineage, browse path, non-embedded datasets

### PR DESCRIPTION
**Pull request by mayurinehate**
**Lineage:**
Remove redundant transitive lineage edges

Before:
before_lineage_fix_pr

After:
after_lineage_fix_pr

**Browse Path**
Fix browse path for custom SQL and embedded datasource

Before:
`f"/{self.config.env.lower()}/{self.platform}/Custom SQL/{csql.get('name', '')}/{csql_id}"`

After
`f"/{self.config.env.lower()}/{self.platform}/{project}/{datasource['name']}/{csql_name}"`

**Non-embedded (tableau-external) datasets**
config added for whether to emit aspects for entities external to tableau (e.g. snowflake/bigquery dataset). default False. 
Source used to always emit external datasets earlier, causing overwrite of aspects from original systems.

**Data Source subtype**
updated subtype for dataset corresponding to PublishedDatasource and EmbeddedDatasource entity to "Published Data source" and "Embedded Data Source" respectively from earlier "Data source" subtype to reduce confusion in lineage view due to datasource with same name.

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/tableau.py [hunk 1]
```diff
@@ -295,22 +324,18 @@ def _create_upstream_table_lineage(
                 type=DatasetLineageTypeClass.TRANSFORMED,
             )
             upstream_tables.append(upstream_table)
-            table_path = f"{project.replace('/', REPLACE_SLASH_CHAR)}/{datasource.get('name', '')}/{table_name}"
+
+            table_path = None
+            if project and datasource.get("name"):
+                table_name = table.get("name") if table.get("name") else table["id"]
+                table_path = f"{project.replace('/', REPLACE_SLASH_CHAR)}/{datasource['name']}/{table_name}"
```
Comment (gabe-lyons) [inline metadata-ingestion/src/datahub/ingestion/source/tableau.py:331]: thanks!

File: metadata-ingestion/src/datahub/ingestion/source/tableau.py [hunk 2]
```diff
@@ -346,7 +371,7 @@ def emit_custom_sql_datasources(self) -> Iterable[MetadataWorkUnit]:
                 custom_sql_connection.get("nodes", [])
             )
             for csql in unique_custom_sql:
-                csql_id: str = csql.get("id", "")
+                csql_id: str = csql["id"]
```
Comment (gabe-lyons) [inline metadata-ingestion/src/datahub/ingestion/source/tableau.py:374]: why this change? what if there is no `id`?

**Review by gabe-lyons — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/tableau.py (hunk 2 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/tableau.py:374]: `id` is mandatory field in `CustomSQLTable` graphql type https://help.tableau.com/current/api/metadata_api/en-us/reference/customsqltable.doc.html
